Common/include/CConfig.hpp

```diff
@@ -2106,22 +2139,133 @@ class CConfig {
 
+  /*!
+   * \brief Get the flame offset (point on plane separating burnt and unburnt zone)
+            for flamelet model initialization.
+   * \return flame offset for flamelet model initialization
+   */
+  su2double *GetFlameOffset(void) { return flame_offset; }
+
+  /*!
+   * \brief Get the flame normal (of the plane separating burnt and unburnt zone)
+            for flamelet model initialization. The normal points in the direction of the burnt zone.
+   * \return flame offset for flamelet model initialization
+   */
+  su2double *GetFlameNormal(void) { return flame_normal; }
+
+  /*!
+   * \brief Get the flame thickness (reaction zone) for flamelet model initialization.
+            This is the thickness of the transition layer from unburnt to burnt conditions.
+   * \return flame thickness for flamelet model initialization
+   */
+  su2double GetFlameThickness(void) { return flame_thickness; }
+
+  /*!
+   * \brief Get the burnt region thickness for flamelet model initialization.
+            This is the thickness of the hot, burnt zone after the reaction zone.
+   * \return flame thickness for flamelet model initialization
+   */
+  su2double GetFlameBurntThickness(void) { return flame_burnt_thickness; }
+
+  /*!
+   * \brief Set the number of scalars for flamelet model.
+   */
+  void SetNScalars(unsigned short n_scalars) { this->n_scalars = n_scalars; }
+
+  /*!
+   * \brief Set the number of controlling variables for flamelet model.
+   */
+  void SetNControlVars(unsigned short n_control_vars) { this->n_control_vars = n_control_vars; }
+
+  /*!
+   * \brief Get the number of control variables for flamelet model.
+   */
+  unsigned short GetNControlVars(void) const { return n_control_vars; }
+
+  /*!
+   * \brief Get the number of transported scalars for flamelet model.
+   */
+  unsigned short GetNScalars(void) const { return n_scalars; }
+
+  /*!
+   * \brief Get the number of user scalars for flamelet model.
+   */
+  unsigned short GetNUserScalars(void) const { return n_user_scalars; }
+
+  /*!
+   * \brief Get the name of the user scalar.
+   */
+  string GetUserScalarName(unsigned short i_user_scalar) const { if(n_user_scalars > 0) return user_scalar_names[i_user_scalar]; else return "NONE"; }
+
+  /*!
+   * \brief Get the name of the user scalar source term.
+   */
+  string GetUserSourceName(unsigned short i_user_source) const { if(n_user_sources > 0) return user_source_names[i_user_source]; else return "NONE"; }
+
+  /*!
+   * \brief Get the number of transported scalars for combustion
+   */
+  unsigned short GetNLookups(void) const { return n_lookups; }
+
+  /*!
+   * \brief Set the total number of LUT sources
+   */
+  void SetNLUTSources(unsigned short n_table_sources) { this->n_table_sources = n_table_sources; }
+
+  /*!
+   * \brief Get the number of transported scalars source terms for combustion
+   */
+  unsigned short GetNLUTSources(void) const { return n_table_sources; }
```

Comment:
Why do you need to get and set so many things into the config? We should not use CConfig as some kind of messenger between other SU2 classes...

Reply:
True. I already removed a couple, I'll go over it again later.

Reply:
I don't know where to move these other functions to. These are all related to the size of the config option list.  Is there a better location for them?

Reply:
If the data is used together you can return a struct with everything.
And ideally you don't need to set anything into the config, just read from it.

Reply:
If I understand correctly, yes. If it needs a "set" try to have it in the class that reads or computes the corresponding value.

Reply:
I removed the Set.. and also 1 Get...